mypy/checker.py

```diff
@@ -240,22 +230,14 @@ def handle_cannot_determine_type(self, name: str, context: Context) -> None:
         else:
             self.msg.cannot_determine_type(name, context)
 
-    def accept(self, node: Union[Expression, Statement, FuncItem],
-               type_context: Type = None) -> Type:
+    def accept(self, stmt: Statement, type_context: Type = None) -> None:
```

Comment:
Statements don't use type context, so the `type_context` argument is redundant now. 

Once the argument is removed, the `type_context` attribute can be moved to `checkexpr.py`.